alns/ALNS.py

```diff
@@ -109,10 +119,63 @@ def add_repair_operator(self, op: _OperatorType, name: str = None):
         name
             Optional name argument, naming the operator. When not passed, the
             function name is used instead.
+        only_after
+            Optional keyword-only argument to indicate after which destroy
+            operators the passed-in repair operator can be used. If None,
+            then it is assumed that all destroy operators can be used.
         """
         logger.debug(f"Adding repair operator {op.__name__}.")
         self._repair_operators[name if name else op.__name__] = op
 
+        if only_after is not None:
+            self._only_after[op].update(only_after)
+
+    def _compute_op_coupling(self) -> np.ndarray:
+        """
+        Internal helper to compute a matrix that describes the
+        coupling between destroy and repair operators. The matrix has size
+        |d_ops| x |r_ops| and entry (i, j) is 1 if destroy operator i can
+        be used in conjunction with repair operator j and 0 otherwise.
+
+        If the only_after keyword argument was not used in when
+        adding repair operators, then all entries of the matrix are 1.
+        """
+        op_coupling = np.zeros(
+            (len(self.destroy_operators), len(self.repair_operators))
+        )
```

Comment:
Suggestion: start assuming the default that all operators work with each other. So `np.ones`, not `np.zeros`.